pkg/nodeupdater/utils.go

```diff
@@ -55,43 +51,31 @@ const (
 // ReadStorageSecretConfiguration ...
 func ReadStorageSecretConfiguration(ctxLogger *zap.Logger) (*StorageSecretConfig, error) {
 	ctxLogger.Info("Fetching secret configuration.")
-	configPath := filepath.Join(config.GetConfPathDir(), configFileName)
-	conf, err := readConfig(configPath, ctxLogger)
+	spObject, err := sp.NewSecretProvider(sp.VPC)
 	if err != nil {
-		ctxLogger.Info("Error loading secret configuration")
+		ctxLogger.Error("Error initializing secret provider", zap.Error(err))
 		return nil, err
 	}
 
-	// Decode g2 API Key if it is a satellite cluster.(unmanaged cluster)
-	if os.Getenv(strings.ToUpper("IKS_ENABLED")) != "True" && os.Getenv(strings.ToUpper("IS_SATELLITE")) == "True" {
-		ctxLogger.Info("Decoding apiKey since its a satellite cluster")
-		apiKey, err := base64.StdEncoding.DecodeString(conf.VPC.G2APIKey)
-		if err != nil {
-			return nil, err
-		}
-		conf.VPC.G2APIKey = string(apiKey)
+	riaasURL, err := spObject.GetRIAASEndpoint(false)
+	if err != nil {
+		ctxLogger.Error("Error fetching RIAAS endpoint", zap.Error(err))
+		return nil, err
 	}
 
 	// Correct if the G2EndpointURL is of the form "http://".
-	conf.VPC.G2EndpointURL = getEndpointURL(conf.VPC.G2EndpointURL, ctxLogger)
-
-	// Correct if the G2TokenExchangeURL is of the form "http://"
-	conf.VPC.G2TokenExchangeURL = getEndpointURL(conf.VPC.G2TokenExchangeURL, ctxLogger)
-
-	riaasInstanceURL, err := url.Parse(fmt.Sprintf("%s/v1/instances?generation=%s&version=%s", conf.VPC.G2EndpointURL, vpcGeneration, vpcRiaasVersion))
+	riaasURL = getEndpointURL(riaasURL, ctxLogger)
+	riaasInstanceURL, err := url.Parse(fmt.Sprintf("%s/v1/instances?generation=%s&version=%s", riaasURL, vpcGeneration, vpcRiaasVersion))
 	if err != nil {
 		ctxLogger.Error("Failed to parse riassInstanceURL", zap.Error(err))
 		return nil, err
 	}
-
 	storageSecretConfig := &StorageSecretConfig{
-		APIKey:              conf.VPC.G2APIKey,
-		IamTokenExchangeURL: fmt.Sprintf("%s/oidc/token", conf.VPC.G2TokenExchangeURL),
-		RiaasEndpointURL:    riaasInstanceURL,
-		BasicAuthString:     fmt.Sprintf("%s:%s", conf.VPC.IamClientID, conf.VPC.IamClientSecret),
+		RiaasEndpointURL: riaasInstanceURL,
 	}
 
-	accessToken, err := storageSecretConfig.GetAccessToken(ctxLogger)
+	// TODO - send appropriate reason for call
```

Comment:
We can tag it as used for VPCNodeLabelUpdator

Reply:
Added reason in GetDefaultIAMToken - "vpc-node-label-updater".